azurerm/resource_arm_virtual_machine_scale_set_test.go

```diff
@@ -690,6 +690,25 @@ func TestAccAzureRMVirtualMachineScaleSet_multipleNetworkProfiles(t *testing.T)
 	})
 }
 
+func TestAccAzureRMVirtualMachineScaleSet_AutoUpdates(t *testing.T) {
```

Comment:
since the upgrade mode can be changed - can we add a separate upgrade test to go from Manual -> Automatic -> Rolling -> Automatic -> Manual (and ensure the `rolling_upgrade_policy` block is set correctly at each stage

Reply:
I'd like to. But `rolling_upgrade_policy` cannot be removed once it is set. This is the same issue as #1310 .

Reply:
@JunyiYi, Does that mean when it goes form `true` -> `false` we just need to force a new resource? I believe this is possible via detecting that change in a `CustomizeDiff` and using `diff.ForceNew()`.

Reply:
It is an API bug (which means it is updatable). You can update from `nil` -> an object of `rolling_upgrade_policy`, but not back. Other fields doesn't matter, they could be updated anytime.

Reply:
Created a BUG in API repository: https://github.com/Azure/azure-rest-api-specs/issues/3751

Reply:
Created a BUG in API repository: https://github.com/Azure/azure-rest-api-specs/issues/3751